### DISCUSSION TITLE
Linking variables with a multiplier at an interface.

### DISCUSSION THREAD

**Discussion by LukeFisherUKAEA** [idaholab/moose#18948] [Q&A Modules: General] [answered]
I am relatively new to MOOSE and am looking to implement a steady-state multiplier at an interface, to link two variables.  Specifically, i'm looking to model a scenario where, for an interface between two regions A and B with variable u in region A and variable v in region B, the boundary condition on v at the interface is a function of the value of variable u at the interface.  There also needs to be a mass flux conservation condition at the boundary, however that can be handled using the InterfaceDiffusionFluxMatch interface kernel.  The function takes other variables, e.g. temperature and volume, and so cannot be implemented using the function kernels, as I understand that these only take variables (x, y, z, t).  As this is an equilibrium situation and we can effectively ignore rate effects, the InterfaceReaction interface kernel is not suitable.

It strikes me that this should be relatively straightforward to do - has anyone any experience of this problem?  Many thanks.

**Specifically, this is to implement Sieverts Law at the interface, where C = K * SQRT( P )
C is the concentration in region B (v) and P is the gas pressure in region A (u).

**Comment by GiudGiud** [comment, marked as answer]
Hello

You may want to consult this discussion https://github.com/idaholab/moose/discussions/18799
to choose between a BC object and an interface kernel with a penalty factor. 

InterfaceReaction could easily be modified to penalize the difference between C and K sqrt(P)

Guillaume

**Reply by LukeFisherUKAEA** [reply]
Hi Guillaume,

Many thanks for this.  Would you be able to point me in the direction of some guidance on modifying kernels, for those that are new to this?

Luke

**Reply by GiudGiud** [reply]
Here's some guidelines for kernels. Interfqace kernels are similar except they are only ran at interface
https://mooseframework.inl.gov/syntax/Kernels/

**Reply by GiudGiud** [reply]
https://mooseframework.inl.gov/syntax/InterfaceKernels/ for interface kernels but it s not as detailed

**Reply by GiudGiud** [reply]
How is this going?

**Reply by LukeFisherUKAEA** [reply]
This has turned out well - with minor modification the combination of InterfaceDiffusion and InterfaceReaction are giving the results I need.  Many thanks for your help.
